src/module.c

```diff
@@ -85,6 +85,17 @@ jl_binding_t *jl_get_binding_wr(jl_module_t *m, jl_sym_t *var)
     return *bp;
 }
 
+// return module of binding
+DLLEXPORT jl_module_t *jl_get_module_of_binding(jl_module_t *m, jl_sym_t *var)
+{
+    jl_binding_t **bp = (jl_binding_t**)ptrhash_bp(&m->bindings, var);
+    if (*bp == HT_NOTFOUND) {
+        jl_errorf("Error: symbol :%s is not bound.", var->name); 
+    }
+    jl_binding_t *b = jl_get_binding((*bp)->owner, var);
+    return b->owner;
+}
+
```

Comment:
@JeffBezanson: I'm not sure this addresses your concerns.  This is the best I could come up with my trial and error approach. If not: explain more, or give me a test case which fails due to the bug, or update it yourself.